### PR TITLE
create livedata module and implement EventLiveData

### PR DESCRIPTION
**Pull request by okuzawats**
create EventLiveData, which is a subtype of MutableLiveData.
that sets the value null after emitting a value.
it is used to notify an event, do not have the value after emitting it.

### CODE REVIEW

**Comment by okuzawats** [comment]
ok finally lint check passed.

File: livedata/src/main/java/jp/co/fuller/yakan/livedata/EventLiveData.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    override fun observe(owner: LifecycleOwner, observer: Observer<in T>) {
+        try {
+            val nullIgnoringObserver = NullIgnoringObserver(observer)
+            super.observe(owner, nullIgnoringObserver)
+        } catch (e: Throwable) {
+            throw e
+        }
+    }
+
+    override fun observeForever(observer: Observer<in T>) {
```
Comment (nanaten) [inline livedata/src/main/java/jp/co/fuller/yakan/livedata/EventLiveData.kt:21]: Is it okay for `removeObserver` not to override?

**Review by nanaten — COMMENTED** [review]
Please check only one point🙏

File: livedata/src/main/java/jp/co/fuller/yakan/livedata/EventLiveData.kt (hunk 1 above)
Comment (okuzawats) [reply livedata/src/main/java/jp/co/fuller/yakan/livedata/EventLiveData.kt:21]: I thought it's ok because nothing special in `removeObserver`.

**Review by nanaten — APPROVED** [review]
LGTM!

**Comment by okuzawats** [comment]
thanks👍

**Pull request merged**